staging/src/k8s.io/apiserver/pkg/admission/interfaces.go

```diff
@@ -49,6 +49,17 @@ type Attributes interface {
 	GetKind() schema.GroupVersionKind
 	// GetUserInfo is information about the requesting user
 	GetUserInfo() user.Info
+	// GetAnnotations returns a copy of saved annotations. Modification to the returned copy of annotations will not be saved to Attributes. Use SetAnnotation/SetAnnotations
+	// to set annotations of Attributes.
+	GetAnnotations() map[string]string
```

Comment:
do we really have to leak the getter here? At all call sites we have a concrete object, not the interface. We don't need that getter in the interface.

Reply:
We can pass an annotation map to `NewAttributesRecord` to avoid this `GetAnnotations`.
In this case annotation would work like other elements. But it needs a lot of modification of codes.

Comments from @tallclair 
> I'm having a hard time tracing this change. I suggest avoiding modifying NewAttributesRecord.